### DISCUSSION TITLE
sempy_labs.admin.list_datasets

### DISCUSSION THREAD

**Discussion by kaem93** [microsoft/semantic-link-labs#361] [General]
I'm trying to execute the command sempy_labs.admin.list_datasets by passing the TokenProvider (through sempy_labs.ServicePrincipalTokenProvider command). 

But when I pass the parameter token_provider I get this error :

datasets = labs.admin.list_datasets(None, None, None, token)

TypeError: list_datasets() takes from 0 to 3 positional arguments but 4 were given

**Comment by m-kovalsky** [comment]
No functions in this library currently support a service principal. That will change soon so stay tuned. Will also provide code examples making it easy to use.

**Comment by m-kovalsky** [comment]
This is now supported in [0.9.0](https://github.com/microsoft/semantic-link-labs/releases/tag/0.9.0).

**Reply by kevchant** [reply]
Will you also be adding service principal support for those passing through an existing bearer token?

For example, those wanting to run the notebooks through the run notebook on demand API?
